shell_commands: make 6Lo compression contexts configurable on non-6LBR

miri64 opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description
This provides the `6ctx` command for not just 6LBR but all 6LoWPAN nodes. For an experimental setup this is really use full if you can't or don't want to rely on compression context dissemination.
<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure
Run `examples/gnrc_networking` on two 6LoWPAN-capable nodes. E.g.

```
BOARD=iotlab-m3 IOTLAB_NODES=2 IOTLAB_SITE=saclay make -C examples/gnrc_networking iotlab-exp iotlab-term
``` 

Configure a global address for each and add a compression context on both for the prefix of the global addresses (note the warning for non-6LBRs that I added). Ping shoud still work as before (maybe even a little bit faster).

```
m3-10;ifconfig 7 add 2001:db8::1
1576852359.881157;m3-10;> ifconfig 7 add 2001:db8::1
1576852359.882008;m3-10;success: added 2001:db8::1/64 to interface 7
m3-11;ifconfig 7 add 2001:db8::2
1576852364.552015;m3-11;> ifconfig 7 add 2001:db8::2
1576852364.552240;m3-11;success: added 2001:db8::2/64 to interface 7
6ctx add 0 2001:db8::/64 120
1576852386.801102;m3-11;> 6ctx add 0 2001:db8::/64 120
1576852386.801412;m3-10;> 6ctx add 0 2001:db8::/64 120
1576852386.801915;m3-10;WARNING: context dissemination by non-6LBR not supported
1576852386.802865;m3-11;WARNING: context dissemination by non-6LBR not supported
m3-10;ping6 2001:db8::2
1576852404.289988;m3-10;> ping6 2001:db8::2
```
<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references
None
<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

miri64 commented:
A keen eye might notice, that I did not provide the actual results in testing setup above. This is because it did not work for some reason when I tried it isolated in this branch :-/. However, in my experiment branch it *does* work... Will investigate, when I find some time.

leandrolanzieri commented:
I tested and works as expected. It even seems to be a decrease in the time as mentioned in the description:
```
m3-10;ping6 2001:db8::11
1578304388.064064;m3-10;> ping6 2001:db8::11
1578304388.079015;m3-10;12 bytes from 2001:db8::11: icmp_seq=0 ttl=64 rssi=-46 dBm time=12.859 ms
1578304389.079464;m3-10;12 bytes from 2001:db8::11: icmp_seq=1 ttl=64 rssi=-46 dBm time=12.852 ms
1578304390.078145;m3-10;12 bytes from 2001:db8::11: icmp_seq=2 ttl=64 rssi=-46 dBm time=12.532 ms
1578304390.078345;m3-10;
1578304390.078945;m3-10;--- 2001:db8::11 PING statistics ---
1578304390.079873;m3-10;3 packets transmitted, 3 packets received, 0% packet loss
1578304390.081859;m3-10;round-trip min/avg/max = 12.532/12.747/12.859 ms

6ctx add 0 2001:db8::/64 120
1578304412.311796;m3-11;> 6ctx add 0 2001:db8::/64 120
1578304412.312861;m3-10;> 6ctx add 0 2001:db8::/64 120
1578304412.314428;m3-11;WARNING: context dissemination by non-6LBR not supported
1578304412.314706;m3-10;WARNING: context dissemination by non-6LBR not supported

m3-10;ping6 2001:db8::11                                                                                                                                 
1578304422.088737;m3-10;> ping6 2001:db8::11
1578304422.101577;m3-10;12 bytes from 2001:db8::11: icmp_seq=0 ttl=64 rssi=-46 dBm time=10.467 ms
1578304423.102770;m3-10;12 bytes from 2001:db8::11: icmp_seq=1 ttl=64 rssi=-46 dBm time=11.428 ms
1578304424.102051;m3-10;12 bytes from 2001:db8::11: icmp_seq=2 ttl=64 rssi=-46 dBm time=10.467 ms
1578304424.102265;m3-10;
1578304424.103835;m3-10;--- 2001:db8::11 PING statistics ---
1578304424.104044;m3-10;3 packets transmitted, 3 packets received, 0% packet loss
1578304424.105405;m3-10;round-trip min/avg/max = 10.467/10.787/11.428 ms
```

miri64 commented:
It still doesn't work for me on current master. @leandrolanzieri what did you do to make it work?

miri64 commented:
Ah, if the routes are set accordingly it works :sweat_smile: 

```
1578572469.601451;m3-11;�main(): This is RIOT! (Version: 2020.01-devel-1629-g0b602)
1578572469.601597;m3-11;RIOT network stack example application
1578572469.602457;m3-11;All up, running the shell now
m3-10;ifconfig 7 add 2001:db8::1
1578572473.341165;m3-10;> ifconfig 7 add 2001:db8::1
1578572473.341451;m3-10;success: added 2001:db8::1/64 to interface 7
m3-11;ifconfig 7 add 2001:db8::2
1578572476.235121;m3-11;> ifconfig 7 add 2001:db8::2
1578572476.235979;m3-11;success: added 2001:db8::2/64 to interface 7
m3-10;nib route add 7 default fe80::54ad:fc65:106b:1115
1578572479.450195;m3-10;> nib route add 7 default fe80::54ad:fc65:106b:1115
m3-11;nib route add 7 default fe80::50bd:fc65:106b:1115
1578572482.556572;m3-11;> nib route add 7 default fe80::50bd:fc65:106b:1115
6ctx add 0 2001:db8::/64 120
1578572486.684493;m3-11;> 6ctx add 0 2001:db8::/64 120
1578572486.685344;m3-11;WARNING: context dissemination by non-6LBR not supported
1578572486.685568;m3-10;> 6ctx add 0 2001:db8::/64 120
1578572486.687745;m3-10;WARNING: context dissemination by non-6LBR not supported
m3-10;ping6 2001:db8::2
1578572490.769528;m3-10;> ping6 2001:db8::2
1578572490.780273;m3-10;12 bytes from 2001:db8::2: icmp_seq=0 ttl=64 rssi=-46 dBm time=8.530 ms
1578572491.778417;m3-10;12 bytes from 2001:db8::2: icmp_seq=1 ttl=64 rssi=-46 dBm time=7.258 ms
1578572492.779175;m3-10;12 bytes from 2001:db8::2: icmp_seq=2 ttl=64 rssi=-46 dBm time=7.886 ms
1578572492.779370;m3-10;
1578572492.780091;m3-10;--- 2001:db8::2 PING statistics ---
1578572492.780667;m3-10;3 packets transmitted, 3 packets received, 0% packet loss
1578572492.783011;m3-10;round-trip min/avg/max = 7.258/7.891/8.530 ms
```

@leandrolanzieri did something similar (adding the global addresses to the neighbor cache) which has the same result.
